### DISCUSSION TITLE
Upgrade?

### DISCUSSION THREAD

**Discussion by gw8674** [tagyoureit/nodejs-poolController#900] [Q&A] [answered]
@rstrouse and @tagyoureit 
First of all, happy belated New Year. Same to everyone else here.

A question or two regarding possibly upgrading my njspC/dashPanel and REM.
I'm still on 7.7.0. I haven't pulled a new njspC, dashPanel or REM in years. Really haven't found a need to. Everything has been running great other than that little crash I had back in November. It seems that all of that was due to that TRENDNet un-managed switch out at the enclosure. It got really wonky, kept disconnecting and was causing me a lot of grief. I replaced it with a TP-Link and haven't had an issue since. 

Anyway, I digress. Like I said, I'm still running much older versions of njspC, dashPanel and REM. My question is this. Do you honestly think there is any good reason for me to upgrade and pull new versions?  I haven't changed any of my equipment (other than that switch) and looking through the commits going back a while, it doesn't seem like those have anything to do with me and my setup. Seems like most are for the Nixie ONLY, non off-the-shelf controller folks.

In this respect, I guess I'm still clinging to the ole adage, (if it aint broke, don't fix it). And it's not broke. Everything is working perfectly and still exceeds what I thought would be possible with all this.

If there is good reasons to upgrade, my understanding is that I will need to upgrade my nodeJS version right? Not even sure what version that is (it's at least a few years old) and I have no idea even the best way to do that.  Also, I am not using PM2. Never got around to it. I still invoke njspC, dashPanel and REM using the old *npm start&* statement in the Command Lines, when and if I need to restart those two Pi's.

Anyway, if you can let me know if there is good reason to upgrade to the latest and greatest and if so, what would be my steps to (somewhat) effortlessly accomplish an upgrade so I don't royally screw everything up beyond all repair (to include my Influx/Grafana pi which is also working great. :)

Thanks much guys and thanks so much for all you have done for me over the years and everyone else here as well.
r.

**Comment by rstrouse** [comment, marked as answer]
The biggest thing will be upgrading your nodeJS version, and PM2.   This can be a bit of an effort since PM2 will continue to try to use the old node version.  Honestly, if it is chugging along without issue I wouldn't touch it for now.  But know if there is a future bug fix it won't be fixed in the old version.

**Reply by gw8674** [reply]
Excellent. Thank you sir...
I'll be sure to keep checking in to see if there is a "commit" that might be applicable to my setup.
Thanks again @rstrouse and take care!
